lib/routes/hanime1/previews.ts

```diff
@@ -0,0 +1,105 @@
+import { Route } from '@/types';
+import { load } from 'cheerio';
+import puppeteer from '@/utils/puppeteer';
+import { config } from '@/config';
+import logger from '@/utils/logger';
+
+export const route: Route = {
+    path: '/previews/:date',
+    name: '新番预告',
+    maintainers: ['kjasn'],
+    example: '/hanime1/previews/202504',
+    categories: ['anime'],
+    parameters: { date: { description: 'Date in YYYYMM format' } },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: true,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['hanime1.me/previews/:date'],
+            target: '/previews/:date',
+        },
+    ],
+    handler: async (ctx) => {
+        const baseUrl = 'https://hanime1.me';
+        const { date } = ctx.req.param();
+        const link = `${baseUrl}/previews/${date}`;
+
+        const browser = await puppeteer();
+        const page = await browser.newPage();
+        await page.setUserAgent(config.ua);
+
+        // 拦截请求
+        await page.setRequestInterception(true);
+        page.on('request', (req) => {
+            const resourceType = req.resourceType();
+            if (['document', 'script', 'xhr', 'fetch', 'image'].includes(resourceType)) {
+                req.continue();
+            } else {
+                req.abort();
+            }
+        });
+
+        // 记录请求
+        logger.http(`Fetching ${link}`);
+        await page.goto(link, {
+            waitUntil: 'domcontentloaded',
+            timeout: 60000, // 设置超时时间为60秒
+        });
+        const response = await page.content();
+        const $ = load(response);
+
+        page.close();
```

Comment:
Unnecessary use of puppeteer. This site works **with ofetch** when using [RSSHub's UA](https://github.com/DIYgod/RSSHub/blob/8508789087d56576430614f6b81de91fdd1f59a7/lib/config.ts#L458).